week1/project/ecommerce/src/components/Categories.jsx

```diff
@@ -0,0 +1,25 @@
+import './Categories.css';
+import Category from './Category'
+
+function Categories ({categories, state, setState}) {
+    
+    function changeFilter(newCategory) {
+        setState(prevState => {
+            return {
+                selectedCategory: newCategory, 
+                isAll: (prevState.selectedCategory === newCategory) && !prevState.isAll
```

Comment:
I think `isAll` always will be false no need for this logic.
or can you explain me this logic maybe I miss understood

Reply:
If I click different categories isAll always will be false.
But if I click the same category (so the first condition is true) then isAll will be inverted.
isAll is indicator either all products should be shown or only products of selected category.
If I will click the same category many times then first time will be shown filtered products, next  time - full list of products, then next time - filtered products and so on.